pyomo/contrib/doe/example/reactor_kinetics.py

```diff
@@ -29,209 +29,199 @@
 
-def disc_for_measure(m, NFE=32):
+def disc_for_measure(m, NFE=32, block=True):
     """Pyomo.DAE discretization
     """
     discretizer = pyo.TransformationFactory('dae.collocation')
-    discretizer.apply_to(m, nfe=NFE, ncp=3, wrt=m.t)
-    for z in m.scena:
-        for t in m.t:
-            m.dCdt_rule[z,'CC',t].deactivate()
+    if block:
+        for s in range(8):
+            discretizer.apply_to(m.block[s], nfe=NFE, ncp=3, wrt=m.block[s].t)
+    else:
+        discretizer.apply_to(m, nfe=NFE, ncp=3, wrt=m.t)
     return m
 
 
-def create_model(scena, const=False, control_time=None, control_val=None, 
-                     t_range=[0.0,1], CA_init=1, C_init=0.1, model_form='dae-index-1', args=[True]):
+def create_model(mod=None, model_option="block", control_time=None, control_val=None, 
+                     t_range=[0.0,1], CA_init=1, C_init=0.1):
     """
     This is an example user model provided to DoE library. 
     It is a dynamic problem solved by Pyomo.DAE.
     
     Arguments
     ---------
-    scena: a dictionary of scenarios, achieved from scenario_generator()
+    model: Pyomo model. If None, a Pyomo concrete model is created
+    model_option: if "parmest", create a process model. if "global", create the global model. 
+        if "block", add model variables and constriants for block.
     control_time: time-dependent design (control) variables, a list of control timepoints
     control_val: control design variable values T at corresponding timepoints
     t_range: time range, h 
     CA_init: time-independent design (control) variable, an initial value for CA
     C_init: An initial value for C
-    model_form: choose from 'ode-index-0' and 'dae-index-1'
-    args: a list, deciding if the model is for k_aug or not. If [False], it is for k_aug, the parameters are defined as Var instead of Param.
-        
+
     Return
     ------
     m: a Pyomo.DAE model 
     """
-    # parameters initialization, results from parameter estimation
-    theta_pe = {'A1': 84.79085853498033, 'A2': 371.71773413976416, 'E1': 7.777032028026428, 'E2': 15.047135137500822}
-    # concentration initialization
-    y_init = {'CA': CA_init, 'CB':0.0, 'CC':0.0}
-    
-    para_list = ['A1', 'A2', 'E1', 'E2']
 
-    if not control_time:
-        control_time = [0,0.125,0.25,0.375,0.5,0.625,0.75,0.875,1]
-        
-    if not control_val:
-        control_val = [300]*9
+    theta = {'A1': 84.79, 'A2': 371.72, 'E1': 7.78, 'E2': 15.05}
 
-    controls = {}
-    for i, t in enumerate(control_time):
-        controls[t]=control_val[i]
-    
-    ### Add variables 
-    m = pyo.ConcreteModel()
-    
-    m.CA_init = CA_init
-    m.para_list = para_list
-    t_control = control_time
-    
-    m.scena_all = scena 
-    m.scena = pyo.Set(initialize=scena['scena-name'])
-    
-    if model_form == 'ode-index-0':
-        m.index1 = False
-    elif model_form == 'dae-index-1':
-        m.index1 = True 
+    if model_option == "parmest":
```

Comment:
Should this be...
if model_option == "parmest":
...
elif model_option == "global":
else:
   if not mod:
     raise error here?

Reply:
This depends on if we want to create the model by ourself, or let user create model in global mode. Either is ok.

Reply:
I think we want to do some error checking to ensure the options passed to `model_option` are valid.

Reply:
We do need to test these value errors are being thrown.